subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c

```diff
@@ -266,8 +267,22 @@ static int os_mgmt_taskstat_read(struct mgmt_ctxt *ctxt)
 /**
  * Command handler: os reset
  */
-static int
-os_mgmt_reset(struct mgmt_ctxt *ctxt)
+static void os_mgmt_reset_work_handler(struct k_work *work)
+{
+	sys_reboot(SYS_REBOOT_WARM);
```

Comment:
This loses the ifdef protection from the cmake file

Reply:
It is under the #ifdef CONFIG_REBOOT, the same used to include the zephyr_os_mgmt.c.